lib/migration.js

```diff
@@ -461,8 +461,8 @@ function mixinMigration(MySQL, mysql) {
 
   function columnType(p, defaultType) {
     var dt = defaultType;
-    if (p.dataType) {
-      dt = String(p.dataType);
```

Comment:
I completely agree with you. Please add/change new/existing test cases in this PR as part of this patch.